### DISCUSSION TITLE
Unhelpful error connecting to Postgres DB

### DISCUSSION THREAD

**Discussion by seajhawk** [Budibase/budibase#7634] [General]
I'm new to Budibase, but I've been doing this tech thing for a long, long time.

My Postgres db is hosted in Azure. It is configured to allow all public access. I am already connecting to it from other applications, albeit ones that are hosted in Azure, and I am able to connect to it from my local computer.

Here's what I did:
- Created a new cloud-hosted instance
- Selected one of the existing templates (Contact List)
- Data > Add Data Source > Postgres
- Entered my connection info
- Got the very unhelpful error message below

I've tried several variations
- different user accounts
- different databases
- SSL enabled/disabled (it is enabled on the Postgres server)

Any thoughts?

![image](https://user-images.githubusercontent.com/18316676/188446384-18b2830a-222c-440e-bb44-24e993708c27.png)

**Comment by andz-bb** [comment]
Hey @seajhawk 

sorry to see that you're having issues connecting to your DB. Could you check if there are any errors coming up in the console or network tab of your browser's devtools (F12 on Chrome)? Hopefully there will be a more detailed error there that will make things clearer!

**Reply by seajhawk** [reply]
Hi @andz-bb - thanks for the quick response!

Here is what I'm seeing in the Console:
![image](https://user-images.githubusercontent.com/18316676/188482771-30a31ae0-b984-48f6-8fda-de0367bb22f9.png)

Network just shows the same 504 Gateway time-out error.

**Reply by andz-bb** [reply]
hey @seajhawk 

I don't have a Postgres server in Azure to test this myself but I'm wondering if this could be a problem with your SSL config here. Do you have a certificate that you can add to the Ca field? [This article might useful here.](https://docs.microsoft.com/en-us/azure/postgresql/single-server/concepts-ssl-connection-security) Hope that helps!

**Reply by seajhawk** [reply]
To provide some more context, I was able to connect to the database from appsmith.com, using Default as the SSL setting. On my Postgres server I have `require_secure_transport` enabled.

I'm not a ssl expert, but I downloaded the .crt file in the Azure article you provided and exported it to a .cer file. Then I copied the public key and pasted it in the CA field of the Postgres connection dialog.
I then tried to connect with ssl enabled, and ssl disabled - neither configuration worked. I got a 504 error in both cases.

**Comment by JayP718** [comment]
Hi,

So I know the Budibase errors that appear on the front end can be a bit vague...

However, sometimes if you see what's printed in your the console. In Chrome (right click - inspect element). A more helpful error message is printed.

Also the logs within the docker container if you are running it within docker are also helpful. If you do "docker ps" to find the id of the container running Budibase then "docker logs (enter id of container) " that should provide a more detailed error.

**Reply by seajhawk** [reply]
Thanks @JayP718 - the Console errors that I replied to Andrew with were the must helpful.

Since I'm running this hosted and not in a container (yet), I don't get any docker logs. But I will keep that in mind in the future.

**Comment by dagelf** [comment]
Seems to be a postgres server version compatibility issue.... Budibase should really support all versions!

**Reply by mike12345567** [reply]
Hi @dagelf - Budibase SQL version support is listed here: https://docs.budibase.com/docs/troubleshooting-sql

Its very difficult for us to support every possible version of every database, due to performance differences and so on. Budibase depends on things like various aggregate support and CTEs.

If there is an error message which isn't particularly helpful/doesn't explain the versions that are supported please let me know the version/database you saw this with.
